include/sparrow_ipc/serialize_utils.hpp

```diff
@@ -0,0 +1,381 @@
+    /**
+     * @brief Serializes a record batch schema into a binary message format.
+     *
+     * This function creates a serialized schema message by combining continuation bytes,
+     * a length prefix, the flatbuffer schema data, and padding to ensure 8-byte alignment.
+     * The resulting format follows the Arrow IPC specification for schema messages.
+     *
+     * @param record_batch The record batch containing the schema to be serialized
+     * @return std::vector<uint8_t> A byte vector containing the complete serialized schema message
+     *         with continuation bytes, 4-byte length prefix, schema data, and 8-byte alignment padding
+     */
+    [[nodiscard]] SPARROW_IPC_API std::vector<uint8_t>
+    serialize_schema_message(const sparrow::record_batch& record_batch);
+
+    /**
+     * @brief Serializes a record batch into a binary format following the Arrow IPC specification.
+     *
+     * This function converts a sparrow record batch into a serialized byte vector that includes:
+     * - A continuation marker at the beginning
+     * - The record batch metadata length (4 bytes)
+     * - The FlatBuffer-encoded record batch metadata containing field nodes and buffer information
+     * - Padding to ensure 8-byte alignment
+     * - The actual data body containing the record batch buffers
+     *
+     * The serialization follows the Arrow IPC stream format where each record batch message
+     * consists of a metadata section followed by a body section containing the actual data.
+     *
+     * @param record_batch The sparrow record batch to be serialized
+     * @return std::vector<uint8_t> A byte vector containing the complete serialized record batch
+     *         in Arrow IPC format, ready for transmission or storage
+     */
+    [[nodiscard]] SPARROW_IPC_API std::vector<uint8_t>
+    serialize_record_batch(const sparrow::record_batch& record_batch);
+
+    template <std::ranges::input_range R>
+        requires std::same_as<std::ranges::range_value_t<R>, sparrow::record_batch>
+    /**
+     * @brief Serializes a collection of record batches into a single byte vector.
+     *
+     * This function takes a range or container of record batches and serializes each one
+     * individually, then concatenates all the serialized data into a single output vector.
+     * The serialization is performed by calling serialize_record_batch() for each record batch
+     * in the input collection.
+     *
+     * @tparam R The type of the record batch container/range (must be iterable)
+     * @param record_batches A collection of record batches to be serialized
+     * @return std::vector<uint8_t> A byte vector containing the serialized data of all record batches
+     *
+     * @note The function uses move iterators to efficiently transfer the serialized data
+     *       from individual record batches to the output vector.
+     */
+    [[nodiscard]] std::vector<uint8_t> serialize_record_batches(const R& record_batches)
```

Comment:
I find the name a bit confusing; at first sight, this method looks very similar to the public API `serialize(cont R& record_batchs)`. Even if the documentation explains the differences in detail, the names sould reflect these difference so that a reader can quickly see these methods are different.